test/e2e/operator_test.go

```diff
@@ -70,172 +76,175 @@ var _ = Describe("Operator API", func() {
 	// 14. Ensure the reference to ns-a is eventually removed from o's status.components.refs field
 	// 15. Delete o
 	// 16. Ensure o is not re-created
-	It("should surface components in its status", func() {
-		o := &operatorsv1.Operator{}
-		o.SetName(genName("o-"))
-
-		Consistently(o).ShouldNot(ContainCopiedCSVReferences())
-
-		Eventually(func() error {
-			return client.Create(clientCtx, o)
-		}).Should(Succeed())
+	Context("when an Operator resource can select its components by label", func() {
```

Comment:
Note: moved this from an `It` to a `Context` so we gain access to BeforeEach/AfterEach/etc. blocks. Before, this It spec would leak testing Operator resources when it failed the test case (but cleaned up properly when it passed).

Reply:
A question and a nit: 

Question: Why not start with a `When`? 

nit: We should probably clean up the long list in the comment too. I believe they were written when old tests were being ported over to ginkgo, but time constraint forced to putting these in `It` blocks and keep those comments for safe measures.

Reply:
@anik120 +1 for both points.

How easily accessible is the same information in those comments if we were to remove them? Only thing I'd want to make sure of is that we're not losing context in remove them.